### PR TITLE
Add `score_hubverse` to codebase

### LINKED ISSUE #171: Add in score_hubverse functionality, with DMB comments addressed
> This file shouldn't really be in this PR at all. Going to remove from this PR and merge it to separately, cleaned up and with your comments addressed

_Originally posted by @dylanhmorris in https://github.com/CDCgov/pyrenew-hew/pull/159#discussion_r1866614843_

### PR DESCRIPTION
**Pull request by dylanhmorris**
Closes #171

### CODE REVIEW

File: pipelines/score_hubverse.R [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    \(x, y) {
+      readr::write_tsv(
+        x,
+        make_output_path(
+          y,
+          "tsv"
+        )
+      )
+    }
+  )
```
Comment (damonbayer) [inline pipelines/score_hubverse.R:146]: this and other formatting seems overly-narrow.

File: pipelines/score_hubverse.R [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+  message("Done.")
+}
+
+p <- arg_parser("Score hubverse tables against observed data.") |>
+  add_argument(
+    "observed_data_path",
```
Comment (damonbayer) [inline pipelines/score_hubverse.R:153]: should we call this `eval_data`?

File: pipelines/score_hubverse.R (hunk 1 above)
Comment (dylanhmorris) [reply pipelines/score_hubverse.R:146]: We can explore tweaking the configuration

File: pipelines/score_hubverse.R (hunk 2 above)
Comment (dylanhmorris) [reply pipelines/score_hubverse.R]: Done in c1ef31d

File: hewr/R/score_hubverse.R [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  )
+
+  scored <- to_score |>
+    scoringutils::score(
+      metrics = c(
+        scoringutils::get_metrics(to_score),
+        list(interval_coverage_95 = interval_coverage_95)
+      )
+    ) |>
+    dplyr::filter(.data$scale == "log")
```
Comment (damonbayer) [inline hewr/R/score_hubverse.R:85]: Will the log scale exist if you don't do some sort of log transform?

Comment (dylanhmorris) [reply hewr/R/score_hubverse.R:85]: Good point. Yikes.

**Review by damonbayer — CHANGES_REQUESTED** [review]
@dylanhmorris I think you need to address my point about the log scale potentially not existing.

File: hewr/R/score_hubverse.R (hunk 3 above)
Comment (dylanhmorris) [reply hewr/R/score_hubverse.R]: Fixed by bf08214

**Comment by dylanhmorris** [comment]
> @dylanhmorris I think you need to address my point about the log scale potentially not existing.

Fixed by bf08214 and ready for re-review @damonbayer

File: hewr/R/score_hubverse.R [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+#' [scoringutils::transform_forecasts()].
+#' @return A table of scores, as the output of
+#' [scoringutils::score()], filtered to include only the
+#' transformed_scale.
+#' @export
+score_hubverse <- function(forecast,
+                           observed,
+                           horizons = c(0, 1),
+                           transform = scoringutils::log_shift,
+                           transform_label = "log",
```
[the remark this answers is not in the document]
Comment (dylanhmorris) [reply hewr/R/score_hubverse.R:45]: Did not know about that flag. Much cleaner, thanks!

[the remark this answers is not in the document]
Comment (dylanhmorris) [reply hewr/R/score_hubverse.R]: Done in 8ff9ec9

**Review by damonbayer — APPROVED** [review]

File: pipelines/score_hubverse.R [hunk 5]
```diff
@@ -67,8 +67,12 @@ score_and_save <- function(observed_data_path,
       show_col_types = FALSE
     ) |>
       dplyr::mutate(disease = !!disease) |>
-      dplyr::filter(.data$target_end_date <= !!last_target_date)
+      dplyr::filter(
+        .data$target_end_date <= !!last_target_date,
+        .data$horizon %in% !!horizons
+      )
 
+    print(to_score)
```
Comment (dylanhmorris) [inline pipelines/score_hubverse.R:75]:
```suggestion
```

**Comment by damonbayer** [comment]
@dylanhmorris would be nice to have a test for 0 length tables

**Pull request merged**